prbt_support/test/acceptance_tests/acceptance_test_joint_limits.py

```diff
@@ -0,0 +1,237 @@
+
+    def _drive_home(self):
+        """ Move the robot to the home position.
+        """
+        positions = self._current_joint_state_positions()
+        diffs = []
+        for i in range(len(positions)):
+            diffs.append(abs(positions[i]-self.home_positions[i]))
+
+        if any(diff > _JOINT_POSITIONS_TOLERANCE for diff in diffs):
+            self._execute_trajectory(self.home_positions)
+
+    def _execute_trajectory(self, positions):
+        """ Execute a single point trajectory given through joint positions (in the order given by self.joint_names). Return true on success and
+            false otherwise.
+        """
+        self.assertEqual(len(positions), len(self.joint_names))
+
+        # determine duration from distance, max velocity and velocity scaling
+        current_positions = self._current_joint_state_positions()
+        durations = []
+
+        for i in range(len(current_positions)):
+            distance = abs(positions[i] - current_positions[i])
+            durations.append(distance/self.joint_velocity_limits[i])
+
+        duration = max(durations)/_VELOCITY_SCALE
+
+        # construct goal
+        traj_point = JointTrajectoryPoint()
+        traj_point.positions = positions
+        traj_point.time_from_start = rospy.Duration(duration)
+
+        traj = JointTrajectory()
+        traj.joint_names = self.joint_names
+        traj.points.append(traj_point)
+
+        goal = FollowJointTrajectoryGoal()
+        goal.trajectory = traj
+
+        # send to trajectory controller
+        self.client.send_goal(goal)
+        self.client.wait_for_result()
+
+        # evaluate result
+        result = self.client.get_result()
+        success = (result.error_code == FollowJointTrajectoryResult.SUCCESSFUL)
+        if not success:
+            rospy.loginfo('Failure executing: ' + str(goal))
+        return success
+
+    def _joint_limit_reaching_test(self, joint_name):
+        """ Test if the robot can be commanded to move exactly to the limits
+
+            Test Sequence:
+              1. Command a movement to the home position.
+              2. Command a movement to the lower limit.
+              3. Command a movement to the upper limit.
+
+            Expected Results:
+              1. Trajectory is executed successfully.
+              2. Trajectory is executed successfully.
+              3. Trajectory is executed successfully.
+        """
+        self._drive_home()
+
+        index = self.joint_names.index(joint_name)
+        limit = _JOINT_LIMITS_DEGREE[joint_name]
+
+        lower_positions = [0] * len(self.joint_names)
+        lower_positions[index] = -radians(limit)
+
+        self.assertTrue(self._execute_trajectory(lower_positions))
+
+        upper_positions = [0] * len(self.joint_names)
+        upper_positions[index] = radians(limit)
+
+        self.assertTrue(self._execute_trajectory(upper_positions))
```

Comment:
I would suggest to put the robot back into home after each test.

Reply:
Done.